File: project-storage-api/src/main/kotlin/mike/dev/project_api/model/Project.kt
package mike.dev.project_api.model

import jakarta.persistence.*
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
@Table(name = "projects")
class Project(
    @Column(nullable = false)
    var name: String,
    @Column(nullable = false)
    var createdBy: String,
    @Column(nullable = false)
    var bpm: Int,
    @Column(nullable = false)
    var quantizeDivision: String,
    @Column(nullable = false)
    var masterVolume: BigDecimal,
    @Column(nullable = false)
    var masterMuted: Boolean = false,
    @OneToMany(mappedBy = "project", cascade = [CascadeType.ALL], orphanRemoval = true)
    var channels: MutableList<Channel> = mutableListOf(),
    @OneToMany(mappedBy = "project", cascade = [CascadeType.ALL], orphanRemoval = true)
    var effects: MutableList<Effect> = mutableListOf(),
    @Column(nullable = false)
    var createdAt: LocalDateTime = LocalDateTime.now(),
    @Column(nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now()
) {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0
} 

File: projects-api-node/src/models/Project.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { Channel } from './Channel.js';
import { Effect } from './Effect.js';

@Entity('projects')
export class Project {
    @PrimaryGeneratedColumn()
    id?: number;

    @Column()
    name?: string;

    @Column()
    createdBy?: string;

    @Column()
    bpm?: number;

    @Column()
    quantizeDivision?: string;

    @Column('decimal', { precision: 10, scale: 2 })
    masterVolume?: number;

    @Column({ default: false })
    masterMuted?: boolean;

    @OneToMany(() => Channel, channel => channel.project, {
        cascade: true,
        onDelete: 'CASCADE'
    })
    channels?: Channel[];

    @OneToMany(() => Effect, effect => effect.project, {
        cascade: true,
        onDelete: 'CASCADE'
    })
    effects?: Effect[];

    @CreateDateColumn()
    createdAt?: Date;

    @UpdateDateColumn()
    updatedAt?: Date;
} 